core-bundle/src/Resources/contao/modules/Module.php

```diff
@@ -275,126 +274,47 @@ protected function getResponseCacheTags(): array
 	 *
 	 * @return string
 	 */
-	protected function renderNavigation($pid, $level=1, $host=null, $language=null)
+	protected function renderNavigation($pid, $level=1, $host=null/*, $language=null*/)
```

Comment:
There is no usage of `$language` in the current version of Module.php anyways.

Reply:
Yes, this seems to be a compatibility leftover from Contao 2. It's not even used in Contao 3.5.40 at least. But, for BC you will need to keep this, otherwise extending from any `Contao\Module` and overriding this protected method will break. I'll incorporate a deprecation in my PR.

Reply:
Actually this is a BC break introduced in Contao `3.5.7`. Prior to that version you were able to override the language - but Contao `3.5.7` introduced the usage of `PageModel::getFrontendUrl`, without carrying this parameter over. Technically we could fix this in 4.9 and thus we would need to continue supporting this, but that's probably not worth it … So I would just deprecate this parameter now and remove it in 5.0 for good.